tests/nested/manual/fde-on-classic/mk-image.sh

```diff
@@ -0,0 +1,328 @@
+    echo -e "ubuntu\nubuntu" | sudo chroot "$DESTDIR" passwd ubuntu
+    echo "ubuntu ALL=(ALL) NOPASSWD:ALL" | sudo tee -a "$DESTDIR"/etc/sudoers
+
+    # XXX set password for root user
+    sudo chroot "$DESTDIR" sh -c 'echo root:root | chpasswd'
+    sudo sh -c "echo \"PermitRootLogin yes\nPasswordAuthentication yes\" >> $DESTDIR/etc/ssh/sshd_config"
+
+    # Populate snapd data
+    cat > modeenv <<EOF
+mode=run
+recovery_system=$SEED_LABEL
+current_recovery_systems=$SEED_LABEL
+good_recovery_systems=$SEED_LABEL
+base=$BASE_SNAP
+gadget=$GADGET_SNAP
+current_kernels=$KERNEL_SNAP
+model=canonical/ubuntu-core-22-pc-amd64
+grade=dangerous
+model_sign_key_id=9tydnLa6MTJ-jaQTFUXEwHl1yRx7ZS4K5cyFDhYDcPzhS7uyEkDxdUjg9g08BtNn
+current_kernel_command_lines=["snapd_recovery_mode=run console=ttyS0 console=tty1 panic=-1 quiet splash"]
+EOF
+    sudo cp modeenv "$DESTDIR"/var/lib/snapd/
+    # needed from the beginning in ubuntu-data as these are mounted by snap-bootstrap
+    # (UC also has base here, but we do not mount it from initramfs in clasic)
+    sudo mkdir -p "$DESTDIR"/var/lib/snapd/snaps/
+    sudo cp "$CACHE/$KERNEL_SNAP" "$CACHE/$GADGET_SNAP" \
+         "$DESTDIR"/var/lib/snapd/snaps/
+    # populate seed
+    local seed_snaps_d="$DESTDIR"/var/lib/snapd/seed/snaps
+    local recsys_d="$DESTDIR"/var/lib/snapd/seed/systems/"$SEED_LABEL"
+    sudo mkdir -p "$recsys_d"/snaps "$recsys_d"/assertions "$seed_snaps_d"
+    if [ -n "$UNASSERTED" ]; then
+        sudo cp "$CACHE"/{pc,pc-kernel,snapd,core22}_*.snap "$recsys_d"/snaps
+    else
+        sudo cp "$CACHE"/{pc,pc-kernel,snapd,core22}_*.snap "$seed_snaps_d"
+    fi
+    sudo cp classic-model.assert "$recsys_d"/model
+    {
+        for assert in "$CACHE"/*.assert; do
+            cat "$assert"
+            printf "\n"
+        done
+    } > "$CACHE"/snap-asserts
+    sudo cp "$CACHE"/snap-asserts "$recsys_d"/assertions/snaps
+    sudo cp model-etc "$recsys_d"/assertions/
+
+    # not needed if we have asserted snaps
+    if [ -n "$UNASSERTED" ]; then
+        cat > "$CACHE"/options.yaml <<EOF
+snaps:
+- name: snapd
+  id: PMrrV4ml8uWuEUDBT8dSGnKUYbevVhc4
+  unasserted: $SNAPD_SNAP
+- name: pc-kernel
+  unasserted: $KERNEL_SNAP
+- name: pc
+  unasserted: $GADGET_SNAP
+- name: core22
+  unasserted: $BASE_SNAP
+EOF
+        sudo cp "$CACHE"/options.yaml "$recsys_d"
+    fi
+}
+
+populate_image() {
+    IMG="$(readlink -f "$1")"
+    CACHE="$(readlink -f "$2")"
+    MNT="$(readlink -f "$3")"
+    local KERNEL_SNAP
+    KERNEL_SNAP=$(find "$CACHE" -maxdepth 1 -name 'pc-kernel_*.snap' -printf "%f\n")
+
+    mkdir -p "$MNT"
+    sudo kpartx -av "$IMG"
+    
+    loop=$(sudo kpartx -l "$IMG" |tr -d " " | cut -f1 -d:|sed 's/..$//'|head -1)
+    loop_esp="${loop}"p2
+    loop_boot="${loop}"p3
+    loop_save="${loop}"p4
+    loop_data="${loop}"p5
+    # XXX: on a real UC device this the ESP is "ubuntu-seed"
+    sudo mkfs.vfat /dev/mapper/"$loop_esp"
```

Comment:
Note that `mkfs.vfat` is a backward compatibility symlink and might disappear one day. The correct name is `mkfs.fat`. See https://github.com/dosfstools/dosfstools/blob/master/src/Makefile.am